From 3c58c35b188d80799948917e1e09721ff8316ad2 Mon Sep 17 00:00:00 2001
From: Davis Goodin <dagood@microsoft.com>
Date: Wed, 31 Oct 2018 22:33:26 -0500
Subject: [PATCH] Skip CoreFx.Private.TestUtilities in source-build

This project referenced the XUnit.Runtime dependency project, causing test-only prebuilt dependencies.

Even though we always pass BuildTests=false, include a condition to handle "true" to make intent clear.
---
 dir.props      | 8 ++++++++
 src/ref.builds | 2 +-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/dir.props b/dir.props
index b75c58047b..530c63c821 100644
--- a/dir.props
+++ b/dir.props
@@ -306,6 +306,14 @@
     <SymbolPackageOutputPath Condition="'$(SymbolPackageOutputPath)'==''">$(PackageOutputPath)symbols/</SymbolPackageOutputPath>
   </PropertyGroup>
 
+  <!--
+    Remove test-only dependencies during source-build: disable project that brings in XUnit.Runtime
+    dependencies through 'ReferenceFromRuntime' property.
+  -->
+  <ItemGroup Condition="'$(DotNetBuildFromSource)' == 'true' and '$(BuildTests)' != 'true'">
+    <ProjectExclusions Include="$(SourceDir)CoreFx.Private.TestUtilities/**/*.csproj" />
+  </ItemGroup>
+
   <!-- Properties related to multi-file mode for ILC tests -->
   <PropertyGroup Condition="'$(TargetGroup)' == 'uapaot'">
     <!-- Only enable multi-file for Release-x64 and Debug-x86 for now -->
diff --git a/src/ref.builds b/src/ref.builds
index c7542be16a..79ee0d40a8 100644
--- a/src/ref.builds
+++ b/src/ref.builds
@@ -7,7 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <Project Include="$(MSBuildThisFileDirectory)*\ref\*.*proj" />
+    <Project Include="$(MSBuildThisFileDirectory)*\ref\*.*proj" Exclude="@(ProjectExclusions)" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.traversal.targets))\dir.traversal.targets" />
 </Project>
\ No newline at end of file
-- 
2.17.1.windows.2

